mirgecom/diffusion.py

```diff
@@ -829,6 +832,9 @@ def diffusion_operator(
     dd_vol_quad = dd_vol.with_discr_tag(quadrature_tag)
     dd_allfaces_quad = dd_vol_quad.trace(FACE_RESTR_ALL)
 
+    # Make sure it has an array context
+    kappa = kappa + dcoll.zeros(array_context=actx, dd=dd_vol)
+
```

Comment:
I added this because in the previous orthotropic cases, I would pass the sub-components as DOFArrays only (i.e., in the driver I would do the `+ zeros`). Now, with this, I don't have to worry about that anymore.

If you prefer, I can add the check if `kappa is an np.ndarray`.

Reply:
Removing this line and passing an array of scalars (see https://github.com/illinois-ceesd/mirgecom/pull/1009/commits/62b8bc009b6953925fd435969eb4b9a175ac6720) fails.
```
    def check_dofarray_against_discr(discr, dof_ary: DOFArray):
        """Verify that the :class:`DOFArray` *dof_ary* is consistent with
        the discretization *discr*, in terms of things like group count,
        number of elements, and number of DOFs per element. If a discrepancy is
        detected, :exc:`InconsistentDOFArray` is raised.
    
        :arg discr: a :class:`~meshmode.discretization.Discretization`
            against which *dof_ary* is to be checked.
        """
        if not isinstance(dof_ary, DOFArray):
>           raise TypeError("non-array passed to check_dofarray_against_discr")
E           TypeError: non-array passed to check_dofarray_against_discr
```

Reply:
Do you have a traceback of where that check occurred?

Reply:
https://github.com/inducer/meshmode/pull/398

Reply:
That solves it. No more changes to `diffusion`